src/main/java/joules/task/Deadline.java

```diff
@@ -0,0 +1,55 @@
+package joules.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import joules.Store;
+
+/**
+ * Represents a deadline task with a specific due date.
+ * A {@code Deadline} extends {@link Task} and adds a {@link LocalDate}
+ * field for the due date. It can be stored and displayed with its
+ * formatted date.
+ */
+public class Deadline extends Task {
+    /** Due date of the deadline task */
+    protected LocalDate by;
```

Comment:
"by" seems like too short of a variable name and is a bit ambiguous. You could change it to something like deadlineDate for clarity
```suggestion
    protected LocalDate deadlineDate;
```